components/switch/tests/unit/switch.spec.tsx

```diff
@@ -0,0 +1,60 @@
+import { GEL } from '@westpac/core';
+import { render, fireEvent } from '@testing-library/react';
+import { screen } from '@testing-library/dom';
+import { Switch } from '@westpac/switch';
+import wbc from '@westpac/wbc';
+import { overridesTest } from '../../../../helpers/tests/overrides-test';
+import { nestingTest } from '../../../../helpers/tests/nesting-test';
+import { SwitchProps } from '../../src/Switch';
+
+overridesTest({
+	name: 'switch',
+	overrides: ['Switch', 'Label', 'Toggle'],
+	Component: ({ label, ...rest }: SwitchProps) => <Switch label="A Switch" {...rest}></Switch>,
+});
+
+nestingTest({
+	name: 'switch',
+	Component: ({ label, ...rest }: SwitchProps) => <Switch label="A Switch" {...rest}></Switch>,
+});
+
+// Component specific tests
+describe('Switch component', () => {
+	const SimpleSwitch = (props: SwitchProps) => (
+		<GEL brand={wbc}>
+			<Switch {...props} />
+		</GEL>
+	);
+
+	test('It should render Switch', () => {
+		const { container } = render(<SimpleSwitch label="A Switch" />);
+		expect(container).toBeInTheDocument();
+	});
+
+	test('should be able to have its checked valued changed', () => {
+		render(<SimpleSwitch label="A Switch" />);
+		const switchInput = screen.getByRole('checkbox');
+		expect(switchInput).toHaveProperty('checked');
+		expect(switchInput).toHaveProperty('checked', false);
+		fireEvent.change(switchInput, { target: { checked: true } });
```

Comment:
could we use `userEvent` from `@testing-library/user-event` instead of fireEvent? as it's more close to actual user's action.
Further explanation here: [Link](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-testing-libraryuser-event)

and make sure that you need to use async and await for userEvent

Reply:
Yes, can do that.  I'll make that change.  Thanks for your input.

Reply:
On top of Soobin's suggestion, you may only need to do fireEvent.click() which I think is the simplest solution if you didn't already try click()

Reply:
userEvent & fireEvent have both been used in the test, just to add some variety, and also to see that both work and have the same effect.  The necessity of async, await, waitFor in the expectations with userEvent makes a bit more work for the userEvent click, so agree with you both: fireEvent simplifies things greatly and just works, while userEvent is very thorough and can test for a lot more things.  I guess they both have their place.